internal/d3l/segments_fetcher_test.go

```diff
@@ -0,0 +1,143 @@
+package d3l_test
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"errors"
+	"github.com/eigerco/strawberry/internal/common"
+	"github.com/eigerco/strawberry/internal/crypto"
+	"github.com/eigerco/strawberry/internal/d3l"
+	"github.com/eigerco/strawberry/internal/erasurecoding"
+	"github.com/eigerco/strawberry/internal/testutils"
+	"github.com/eigerco/strawberry/internal/work"
+	"github.com/eigerco/strawberry/pkg/network/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+type assurerClientMock struct {
+	mock.Mock
+}
+
+func (a *assurerClientMock) GetAllPeers() []*peer.Peer {
+	args := a.MethodCalled("GetAllPeers")
+	return args.Get(0).([]*peer.Peer)
+}
+
+func (a *assurerClientMock) SegmentShardRequestSend(ctx context.Context, peerKey ed25519.PublicKey, erasureRoot crypto.Hash, shardIndex uint16, segmentIndexes []uint16) (segmentShards [][]byte, err error) {
+	args := a.MethodCalled("SegmentShardRequestSend", ctx, peerKey, erasureRoot, shardIndex, segmentIndexes)
+	return args.Get(0).([][]byte), args.Error(1)
+}
+
+func TestNewSegmentsFetcher(t *testing.T) {
+	assurers := make([]*peer.Peer, common.NumberOfValidators)
```

Comment:
consider adding the build tag for tiny configuration to avoid slowing down the pipeline

Reply:
I am not sure if we should do this, we already use tiny configuration for the e2e test, I think we should have at least a unit test that tests with the full configuration